src/layers/core/path-layer/path-layer.js

```diff
@@ -97,7 +82,11 @@ export default class PathLayer extends Layer {
   }
 
   calculateIndices(attribute) {
-    const {paths, IndexType, model, indexCount} = this.state;
+    const {paths, IndexType, model, pointCount} = this.state;
+
+    // each path with length n has n-1 line segments
+    // * 2 * 3: each segment is rendered as 2 triangles with 3 vertices
+    const indexCount = (pointCount - paths.length) * 2 * 3;
 
     if (IndexType === Uint16Array && indexCount > 65535) {
```

Comment:
It's not the indexCount but the vertexCount you need to check (i.e. an Uint16Array can have more than 65K elements, but each value can only be max 65K).